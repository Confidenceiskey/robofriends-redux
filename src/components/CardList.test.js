import React from 'react';
import { shallow } from 'enzyme';
import CardList from './CardList';

it('expect to render CardList component', () => {
  const mockRobots = [
    {
      id: 'K78N2P9',
      name: 'John Doe',
      email: 'doemasterjohn@john.com'
    },
    {
      id: 'L34J1O5',
      name: 'Sally Jones',
      email: 'sallysmells@sallyconsulting.com'
    }
  ];

  const wrapper = shallow(
    <CardList robots={mockRobots} />
  );

  expect(wrapper).toMatchSnapshot();
});