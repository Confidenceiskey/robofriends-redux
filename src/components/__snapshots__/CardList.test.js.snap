// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    robots={
      Array [
        Object {
          "email": "doemasterjohn@john.com",
          "id": "K78N2P9",
          "name": "John Doe",
        },
        Object {
          "email": "sallysmells@sallyconsulting.com",
          "id": "L34J1O5",
          "name": "Sally Jones",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          email="doemasterjohn@john.com"
          id="K78N2P9"
          name="John Doe"
        />,
        <Card
          email="sallysmells@sallyconsulting.com"
          id="L34J1O5"
          name="Sally Jones"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "email": "doemasterjohn@john.com",
          "id": "K78N2P9",
          "name": "John Doe",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "email": "sallysmells@sallyconsulting.com",
          "id": "L34J1O5",
          "name": "Sally Jones",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            email="doemasterjohn@john.com"
            id="K78N2P9"
            name="John Doe"
          />,
          <Card
            email="sallysmells@sallyconsulting.com"
            id="L34J1O5"
            name="Sally Jones"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "email": "doemasterjohn@john.com",
            "id": "K78N2P9",
            "name": "John Doe",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "email": "sallysmells@sallyconsulting.com",
            "id": "L34J1O5",
            "name": "Sally Jones",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
